src/main/java/com/fasterxml/jackson/databind/ObjectReader.java

```diff
@@ -354,7 +354,11 @@ protected JsonToken _initForReading(JsonParser p) throws IOException
             t = p.nextToken();
             if (t == null) {
                 // Throw mapping exception, since it's failure to map, not an actual parsing problem
-                throw JsonMappingException.from(p, "No content to map due to end-of-input");
+                if (this._config.hasDeserializationFeatures(DeserializationFeature.READ_NULL_AS_EMPTY_COLLECTION.getMask())) {
```

Comment:
Not sure about this one: note that this does not mean JSON `null` token, but actual physical end-of-input.
So I don't think it should fall under this feature; another feature could of course allow consider end-of-input as null equivalent, and thereby return either `null` or empty collection or whatever other coercions might be applicable.